src/main/java/Calculator.java

```diff
@@ -1,74 +1,34 @@
+/**
+ * This interface is used to calculate valid moves and attack coverage for a chess piece on a chess board.
+ * Used in classes that represent specific types of chess pieces (Pawn, Rook, Knight, Bishop, Queen, King)
+ * The group should be implementing this interface and provide appropriate implementations for the methods.
+ * Note: the Calculator assumes that the player is not in Check. (CheckmateCalculator will be responsible for that.)
```

Comment:
Checkmate Calculator will just check for checkmate.

The use case for choosing a piece and displaying valid moves will handle that.